controller_manager:
  ros__parameters:
    update_rate: 100  # Hz - controller update frequency
    
    # List of controllers to load
    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster
    
    position_controller:
      type: position_controllers/JointGroupPositionController
    
    trajectory_controller:
      type: joint_trajectory_controller/JointTrajectoryController

joint_state_broadcaster:
  ros__parameters: {}

position_controller:
  ros__parameters:
    joints:
      - joint1
      - joint2
      - joint3
    
    command_interfaces:
      - position
    
    state_interfaces:
      - position
      - velocity

trajectory_controller:
  ros__parameters:
    joints:
      - joint1
      - joint2
      - joint3
    
    command_interfaces:
      - position
    
    state_interfaces:
      - position
      - velocity
    
    constraints:
      stopped_velocity_tolerance: 0.01
      goal_time: 0.5
      
      joint1:
        trajectory: 0.05
        goal: 0.02
      joint2:
        trajectory: 0.05
        goal: 0.02
      joint3:
        trajectory: 0.05
        goal: 0.02
    
    state_publish_rate: 50.0
    action_monitor_rate: 20.0
    allow_partial_joints_goal: false
    allow_integration_in_goal_trajectories: true
    open_loop_control: false
    interpolation_method: splines
